sdk/core/Azure.Core/tests/HttpClientTransportFunctionalTest.cs

```diff
@@ -33,5 +36,31 @@ protected override HttpPipelineTransport GetTransport(bool https = false)
             }
             return new HttpClientTransport();
         }
+
+#if NETCOREAPP
+        [Test]
+        public void UsesSocketsClientTransportOnNetCoreApp()
+        {
+            var transport = (HttpClientTransport)GetTransport();
+            HttpClient transportClient = transport.Client;
+            var handler = (SocketsHttpHandler)GetHandler(transportClient);
+            Assert.AreEqual(TimeSpan.FromSeconds(300), handler.PooledConnectionLifetime);
+            Assert.GreaterOrEqual(handler.MaxConnectionsPerServer, 50);
+        }
+#else
+        [Test]
+        public void UsesHttpClientTransportOnNetStandart()
+        {
+            var transport = (HttpClientTransport)GetTransport();
+            HttpClient transportClient = transport.Client;
+            var handler = (HttpClientHandler)GetHandler(transportClient);
+            Assert.AreEqual(50, handler.MaxConnectionsPerServer);
+        }
+#endif
+
+        private static object GetHandler(HttpClient transportClient)
```

Comment:
No a fan of this myself but want to make sure nothing is broken during refactorings.